lib/req_ch.ex

```diff
@@ -73,25 +70,15 @@ defmodule ReqCh do
   defp run(%Req.Request{private: %{clickhouse_format: _}} = request), do: request
 
   defp run(%Req.Request{options: %{clickhouse: _query}} = request) do
-    url_parts = [
-      get_option(request, :scheme, "http"),
-      "://",
-      maybe_credentials(request),
-      get_option(request, :hostname, "localhost"),
-      ":",
-      get_option(request, :port, "8123")
-    ]
-
-    url =
-      url_parts
-      |> IO.iodata_to_binary()
-      |> URI.parse()
+    request =
+      with %Req.Request{options: %{} = opts} when not is_map_key(opts, :base_url) <- request do
+        Req.Request.merge_options(request, base_url: "http://localhost:8123")
+      end
```

Comment:
nitpick: we can also do a one liner:

```suggestion
    request = update_in(request.options, &Map.put_new(&1, :base_url, "http://localhost:8123"))
```

I'll add it upstream: https://github.com/wojtekmach/req/issues/432.